mne_bids/write.py

```diff
@@ -1072,4 +1073,59 @@ def write_anat(bids_root, subject, t1w, session=None, acquisition=None,
         fname = t1w_basename.replace('.nii.gz', '.json')
         _write_json(fname, t1w_json, overwrite, verbose)
 
+        if deface is not None:
+            t1w_data_defaced = _deface_t1w(t1w.get_data(), deface,
+                                           mri_landmarks)
+            t1w = nib.Nifti1Image(t1w_data_defaced, t1w.affine, t1w.header)
+
+    # Save anatomical data
+    if op.exists(t1w_basename):
+        if overwrite:
+            os.remove(t1w_basename)
+        else:
+            raise IOError('Wanted to write a file but it already exists and '
+                          '`overwrite` is set to False. File: "{}"'
+                          .format(t1w_basename))
+
+    nib.save(t1w, t1w_basename)
+
     return anat_dir
+
+
+def _deface_t1w(t1w_data, method, mri_landmarks, theta=35):
+    # x: L/R L+, y: S/I I+, z: A/P A+
+    normal = np.cross(mri_landmarks[0], mri_landmarks[1])
+    normal /= np.linalg.norm(normal)
+    deface_indices = np.zeros(t1w_data.shape, dtype=bool)
+    for i in range(t1w_data.shape[0]):
+        for j in range(t1w_data.shape[1]):
+            for k in range(t1w_data.shape[2]):
+                # translate to NAS, and shift up
+                v = np.array([i, j, k]) - mri_landmarks[1]
```

Comment:
better to avoid single letter variable letters except for indices. They are unintelligible and harder to grep